tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -452,11 +465,27 @@ export class WebGPUBackend extends KernelBackend {
 
     const encoder = this.device.createCommandEncoder();
     const pass = encoder.beginComputePass();
+    if (shouldTimeProgram) {
+      pass.writeTimestamp(this.querySet, this.cc); // --ForQuery
+    }
     pass.setPipeline(pipeline);
     pass.setBindGroup(0, bg);
     pass.dispatch(
         program.dispatch[0], program.dispatch[1], program.dispatch[2]);
+    if (shouldTimeProgram) {
+      pass.writeTimestamp(this.querySet, this.cc+1); // --ForQuery
+    }
     pass.endPass();
+
+    if (shouldTimeProgram) {
+    //  encoder.resolveQuerySet(this.querySet, this.cc, 2, this.dstBuffer, 8*this.cc); // --ForQuery
+      if(this.cc === (this.nn-1)*2) {
+        encoder.resolveQuerySet(this.querySet, 0, this.cc, this.dstBuffer, 0); // --ForQuery
+      }
+      ++this.cc; // --ForQuery
+      ++this.cc; // --ForQuery
```

Comment:
It seems that above two lines should be put before L482. For example, if you have two querySet. In line 483, you want to call `encoder.resolveQuerySet(this.querySet, 0, 2, this.dstBuffer, 0);`. However, in your case, this.cc is still 0 at that time.